datadog_checks_base/datadog_checks/base/checks/win/wmi/sampler.py

```diff
@@ -400,15 +400,15 @@ def build_where_clause(fltr):
             while f:
                 prop, value = f.popitem()
 
-                if isinstance(value, (tuple, list)):
+                if isinstance(value, (tuple, list)) and len(value) == 2:
```

Comment:
I don't think this solves the issue. IIUC, the errors comes up if value is a list of one element.
This can only happen with the win32_event_log check in cases like this https://github.com/DataDog/integrations-core/blob/cc324e83c2fedad3267f70ced9f09ef935ce51cf/win32_event_log/datadog_checks/win32_event_log/win32_event_log.py#L102

We have
```
value = [
    ('=', 'foo'),
    ('=', 'bar')
    ...
]
```
I don't think that checking the length of value is the right thing to do otherwise we'll end up with `oper = ('=', 'foo')` and `value = ('=', 'bar')`

Reply:
added a specific case on the new test and another hack to the pile